bin/i18n/redact_restore_utils.rb

```diff
@@ -57,70 +68,54 @@ def self.restore_data(source_data, redacted_data, plugins = [], format = 'md')
     source_json.close
     redacted_json.close
 
-    return restored
+    return JSON.parse(restored)
   end
 
   def self.restore(source, redacted, dest, plugins = [], format = 'md')
     return unless File.exist?(source)
     return unless File.exist?(redacted)
-    is_json = File.extname(source) == '.json'
-    source_data =
-      if is_json
-        JSON.parse(File.read(source))
-      else
-        YAML.load_file(source)
-      end
-    redacted_data =
-      if is_json
-        JSON.parse(File.read(redacted))
-      else
-        YAML.load_file(redacted)
-      end
 
-    return unless source_data
-    return unless redacted_data
-    return unless source_data&.values&.first&.length
-    return unless redacted_data&.values&.first&.length
+    if I18nScriptUtils.json_file?(source) || I18nScriptUtils.yaml_file?(source)
+      source_data = I18nScriptUtils.parse_file(source)
+      redacted_data = I18nScriptUtils.parse_file(redacted)
```

Comment:
`I18nScriptUtils.parse_file` contains the logic to parse `yaml` or `json` files.